lib/semantics/scope.cc

```diff
@@ -48,6 +48,18 @@ Scope::size_type Scope::erase(const SourceName &name) {
     return 0;
   }
 }
+Scope *Scope::FindSubmodule(const SourceName &name) const {
+  auto it{submodules_.find(name)};
+  if (it == submodules_.end()) {
+    return nullptr;
+  } else {
+    return it->second;
+  }
+}
+Scope *Scope::AddSubmodule(const SourceName &name, Scope *submodule) {
+  auto pair{submodules_.emplace(name, submodule)};
+  return !pair.second ? pair.first->second : nullptr;
```

Comment:
What's being returned here?  A pointer to an extant Scope for a submodule of the same name that was already present in the `submodules_` map for the scope?

Reply:
Yes, but I'm only using it as a boolean now so I'll change it to return `pair.second`. I.e. `true` if the insertion took place.